packages/zowe-explorer/__tests__/__unit__/trees/shared/SharedInit.unit.test.ts

```diff
@@ -392,6 +393,64 @@ describe("Test src/shared/extension", () => {
             await SharedInit.watchConfigProfile(context, { ds: "ds", uss: "uss", job: "job" } as any);
             expect(spyRefreshAll).toHaveBeenCalled();
         });
+
+        it("should trigger callbacks when credentials or credManager are updated by another application", async () => {
+            let onVaultUpdatedCallback, onCredentialManagerUpdatedCallback;
+            const dummyWatcher: any = {
+                subscribeUser: (_event, cb) => {
+                    onVaultUpdatedCallback = cb;
+                    return { close: () => {} } as any;
+                },
+                subscribeShared: (_event, cb) => {
+                    onCredentialManagerUpdatedCallback = cb;
+                    return { close: () => {} } as any;
+                },
+            };
+            const spyWatcher = jest.spyOn(imperative.EventOperator, "getWatcher").mockReturnValue(dummyWatcher);
+            const spyGuiError = jest.spyOn(Gui, "errorMessage");
+
+            // Spy callback behavior
+            const spyTranslatedLog = jest.spyOn(vscode.l10n, "t");
+            const spyGetProfileInfo = jest.spyOn(profUtils.ProfilesUtils, "getProfileInfo").mockImplementationOnce(jest.fn());
+            const spyReadConfigFromDisk = jest.spyOn(profUtils.ProfilesUtils, "readConfigFromDisk").mockImplementationOnce(jest.fn());
+            const spyRefreshAll = jest.spyOn(SharedActions, "refreshAll").mockImplementation(jest.fn());
+
+            // Setup watchers
+            await SharedInit.watchConfigProfile(context, { ds: "ds", uss: "uss", job: "job" } as any);
```

Comment:
After merging in changes from next branch, the `providers` argument can be removed here 😋